validate/validate.go

```diff
@@ -457,22 +501,16 @@ func envValid(env string) bool {
 	return true
 }
 
-func capValid(capability string) bool {
-	for _, val := range defaultCaps {
-		if val == capability {
-			return true
-		}
-	}
-	return false
-}
-
-func rlimitValid(rlimit string) bool {
+func rlimitValid(rlimit rspec.Rlimit) error {
 	for _, val := range defaultRlimits {
-		if val == rlimit {
-			return true
+		if val == rlimit.Type {
+			if rlimit.Hard < rlimit.Soft {
+				return fmt.Errorf("hard limit of rlimit %s should not be less than soft limit", rlimit.Type)
+			}
```

Comment:
I'd rather not bury the hard/soft comparison inside the type check.  I'm fine checking this before the type, because “hard < soft” is *always* wrong, but “type not in `defaultRlimits`” could be “runtime-tools maintainers haven't taught runtime-tools about the new rlimit”.  If you really want the type check first, you could do something like:

    found := false
    for … {
      if val == rlimit.Type {
        found = true
        break
      }
    }
    if !found {
      return fmt.Errorf("rlimit type %q is invalid", rlimit.Type)
    }